tests/filesystem.go

```diff
@@ -58,7 +59,7 @@ func prepareRootPartitionForPasswd(ctx context.Context, root *types.Partition) (
 		return err
 	}
 	defer func() {
-		err = umountPartition(root)
+		err = errors.Join(err, umountPartition(root))
```

Comment:
tests/filesystem: Join cleanup errors

Would be nice to include which lint we are fixing by using join.